### ISSUE TITLE
HOMME not b4b over multiple PE layouts with cubed_sphere_map=2

### ISSUE THREAD

**Issue by mrnorman** [E3SM-Project/E3SM#2714]
The theta-l dycore in HOMME is not b4b when you run multiple PE layouts. If I run on Summit with 42 tasks per node, it is b4b from run to run. However, that configuration is not b4b with 18 MPI tasks per node with 2 threads per task. I'm happy to help identify the offending code, but I wanted to have this documented on github at least. Namelist for this experiment is below:

```
&ctl_nl
ndays = 1
NThreads=-1   ! use maximum threads available
partmethod    = 4
topology      = "cube"
test_case     = "asp_baroclinic"
u_perturb = 1
rotate_grid = 0
ne=4
qsize = 10
nmax = 360
statefreq=360
restartfreq   = 43200
restartfile   = "./R0001"
runtype       = 0
tstep=40
rsplit=3
qsplit = 1
tstep_type = 5
integration   = "explicit"
nu=4.5e17
nu_div=4.5e17
nu_p=4.5e17
nu_q=4.5e17
nu_s=4.5e17
nu_top = 0e5
se_ftype     = 0
limiter_option = 9
vert_remap_q_alg = 1
hypervis_order = 2
hypervis_subcycle=1
hypervis_subcycle_q=1
theta_hydrostatic_mode = .true.
/
```

**Comment by rljacob** [comment]
Do you mean 2 runs at 18 aren't bfb?  Or 18 and 42 are not bfb with each other?

**Comment by mrnorman** [comment]
(18 tasks, 2 threads per task) is not b4b with (42 tasks, 1 thread per task). Each is b4b with itself across multiple runs of the same config, but this would fail ERS when integrated into E3SM, I think.

**Comment by mrnorman** [comment]
Well, not ERS, but the one where we change the PE layout, I mean

**Comment by mt5555** [comment]
@oksanaguba has seen this also with PREQX, with some combinations of MPI tasks. For intel compilers, I think it turned out to be  sensitive to the -fp-model option.  IBM might have a similar issue.

And of course it's possible the theta-l model has introduced a non-reproducible MPI issue.

**Comment by worleyph** [comment]
@mrnorman , is the 42 task job built with FORCE_BUILD_SMP=TRUE? Might be worthwhile comparing two MPI-only experiments or two experiments both with threading (or setting FORCE_BUILD_SMP to true and trying again).

**Comment by mrnorman** [comment]
This is with standalone HOMME, and both instances are built with the -mp flag if that's what you mean.

**Comment by worleyph** [comment]
Yes. Thanks.

**Comment by oksanaguba** [comment]
Can you tell from the output which cubed_sphere_map is used? I think it was on about that. With preqx, the default map is 0 and you may have namelists with map 2. Map 2 won't be b4b but map 0 should be. You can also change it in the namelist and see whether with map 0 it goes away (or I will do this at some point, it does not have to be summit).

**Comment by oksanaguba** [comment]
By the way with preqx map 2 has this behavior, too. It is not about threads, this comes up from different mpg layouts.

**Comment by mrnorman** [comment]
My namelist doesn't specify cubed_sphere_map. Do I add this to the ctl_nl?

**Comment by oksanaguba** [comment]
Yes, I just looked and theta-l by default uses map 2. Change it in ctr_nl with  cubed_sphere_map=0 .

**Comment by mt5555** [comment]
Lets reopen this issue, since we are transitioning to cubed_sphere_map=2 for all future simulations ( when we transition to TempestRemap mapping files).  So this needs to be fixed at some point before v2.

**Comment by mrnorman** [comment]
Cool. Is it OK if I go ahead with map=0 for my GPU porting? Will it affect anything substantial?

**Comment by mt5555** [comment]
Yes, cubed_sphere_map=0 is fine for standalone HOMME simulations.

**Comment by oksanaguba** [comment]
Found an old issue #2535 , but search by issues is not good, it did not come up, I had to scan the list of issues.

**Comment by mt5555** [comment]
fixed by E3SM-Project/E3SM#2824
